packit/base_git.py

```diff
@@ -156,17 +163,21 @@ def commit(self, title: str, msg: str, prefix: str = "[packit] ") -> None:
                 "No changes are present in the dist-git repo: nothing to commit."
             )
         self.local_project.git_repo.index.write()
-        commit_args = ["-s", "-m", main_msg]
-        if msg:
-            commit_args += ["-m", msg]
-        # TODO: attach git note to every commit created
-        # TODO: implement cleaning policy: once the PR is closed (merged/refused), remove the branch
-        #       make this configurable so that people know this would happen, don't clean by default
-        #       we should likely clean only merged PRs by default
-        # TODO: implement signing properly: we need to create a cert for the bot,
-        #       distribute it to the container, prepare git config and then we can start signing
-        # TODO: make -s configurable
-        self.local_project.git_repo.git.commit(*commit_args)
+
+        with tempfile.NamedTemporaryFile(mode="w+t") as fp:
+            fp.writelines([main_msg, "\n"])
+            if msg:
+                fp.writelines(["\n", msg, "\n"])
+            fp.seek(0)
+            args = ["--in-place"]
+            for token, value in trailers or []:
+                args += ["--trailer", f"{token}={value}"]
+            if trailers:
+                self.local_project.git_repo.git.interpret_trailers(*args, fp.name)
+
+            # TODO: make -s configurable
+            commit_args = ["-s", "-F", fp.name]
+            self.local_project.git_repo.git.commit(*commit_args)
```

Comment:
This chunk is pretty impressive and thank you for cleaning up the outdated TODO list.